src/lib/bootstrap.js

```diff
@@ -35,5 +35,22 @@ WebComponents.waitFor(async () => {
 });
 
 if ("serviceWorker" in navigator) {
-  navigator.serviceWorker.register("/sw.js");
+  const allowedHostnames = ["www.web.dev", "localhost"];
+  if (allowedHostnames.indexOf(window.location.hostname) !== -1) {
+    navigator.serviceWorker.register("/sw.js");
+  } else {
+    console.warn(
+      "skipping SW, unsupported hostname:",
+      window.location.hostname,
+    );
+
+    // Remove previous Service Worker instances from this hostname. This should never normally
+    // happen but is here for safety.
+    navigator.serviceWorker.getRegistrations().then(async (all) => {
+      await all.map((reg) => reg.unregister());
+      if (all.length) {
+        window.location.reload();
```

Comment:
Because this does a reload do you think folks would actually see the warning on 42?

Reply:
Well, only if they had a SW installed already. This isn't likely because e.g., appengine-test is never going to see this code (the SW will replace itself before new JS is brought in).

This whole branch only exists in case we mess up again and serve the content from a weird domain.